subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/ValidationActions.java

```diff
@@ -35,7 +38,11 @@ public void doValidate(String propertyName, Object value, TaskValidationContext
         public void doValidate(String propertyName, Object value, TaskValidationContext context) {
             File file = toFile(context, value);
             if (!file.exists()) {
-                context.visitError(String.format("File '%s' specified for property '%s' does not exist.", file, propertyName));
+                if (Files.isSymbolicLink(file.toPath())) {
```

Comment:
Like stated in the test, I don't think we should change the validation. We should handle non-existing files and broken symlinks in the same way, so we need to fail here.

Reply:
Hmm, did I misunderstand what @lptr requested here?

Reply:
What is it that @lptr requested? Today we talked about using `@InputFiles` instead of `@InputFile` since we shouldn't need to adjust the validation. Do I remember that correctly?

Reply:
Thanks for clarifying, reverted the change, kept only the passing test